File: Jump_to_Algorithm3/week14/SoohyunKim/9935_문자열폭발.js
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');

function solution(input) {
  const str = input[0];
  const target = input[1];
  const len = target.length;
  const stack = [];

  for (let i = 0; i < str.length; i++) {
    stack.push(str[i]); // 스택 추가
  
    // 마지막 문자가 target과 일치할 경우
    if (stack.length >= len && stack.slice(-len).join('') === target) {
      stack.splice(-len, len); // 폭발 문자열 제거
    }
  }
  const result = stack.join("");
  return result === "" ? "FRULA" : result;
}

console.log(solution(input));

File: Jump_to_Algorithm3/week14/SeoyeonHong/9935_문자열폭발.py
# 모든 폭발이 끝난 후 남은 문자열 출력
string = input()
exp = list(input()) # 폭발 문자열
l = len(exp)
result = [] # 문자열 조작이 많으므로 문자열 대신 리스트 사용

for s in string:
    result.append(s)
    if len(result) >= l and result[-l:] == exp: # 폭발 문자열일 경우
        del result[-l:] # 리스트 구간 삭제

print(''.join(result)) if result else print("FRULA")


# 시간초과
# string = input()
# exp = input()

# while exp in string:
#     string = string.replace(exp, '') # 새로운 문자열을 생성해 할당하므로 시간이 오래 걸림

# print(string) if string else print("FRULA")